src/module/mod.rs

```diff
@@ -256,9 +256,17 @@ impl Module {
         self.funcs.emit(&mut cx);
         self.data.emit(&mut cx);
 
-        emit_name_section(&mut cx);
+        if self.config.generate_names {
+            emit_name_section(&mut cx);
+        }
+
         self.producers.emit(&mut cx);
         for section in self.custom.iter() {
+            if !self.config.generate_dwarf && section.name.starts_with(".debug") {
```

Comment:
I wonder if this should also be skipped when parsing to avoid keeping the dwarf sections in memory?